khard/carddav_object.py

```diff
@@ -333,6 +339,71 @@ def anniversary(self):
                 pass
         return None
 
+    def _get_ablabel(self, item):
+        """Get an ABLABEL for a specified item in the vCard.
+        Will return the ABLABEL only if the item is part of a group with exactly
+        two items, exactly one of which is an ABLABEL.
+
+        :param item: the item to be labelled
+        :type item: vobject.base.ContentLine
+        :returns: the ABLABEL in the circumstances above or an empty string
+        :rtype: str
+
+        """
+        label = ""
+        if item.group:
+            count = 0
+            for child in self.vcard.getChildren():
+                if child.group and child.group == item.group:
+                    count += 1
+                    if child.name == "X-ABLABEL":
+                        if label == "":
+                            label = child.value
+                        else:
+                            return ""
+            if count != 2:
+                label = ""
+        return label
+
+    def _get_new_group(self, group_type=""):
+        """Get an unused group name for adding new groups. Uses the form item123
+         or itemgroup_type123 if a grouptype is specified.
+
+        :param group_type: (Optional) a string to add between "item" and the
+                           number
+        :type group_type: str
+        :returns: the name of the first unused group of the specified form
+        :rtype: str
+
+        """
+        counter = 1
+        while True:
+            group_name = "item%s%d" % (group_type, counter)
+            for child in self.vcard.getChildren():
+                if child.group and child.group ==  group_name:
+                    counter += 1
+                    break
+            else:
+                return group_name
+
+    def _add_labelled_object(self, obj_type, user_input, name_groups=False):
+        obj = self.vcard.add(obj_type)
+        if isinstance(user_input, dict):
+            if len(user_input) > 1:
+                raise ValueError("Error: %s must be a string or a dict " +\
+                                 "containing one key/value pair." % obj_type)
+            label = [i for i in user_input][0]
```

Comment:
Sorry it is `list(user_input)[0]` not `user_input.keys()[0]` as I previously suggested. The list comprehension `[x for x in y]` just looks like a noop so `list(y)` might be more comprehensible.

Reply:
Ah, type casting :) In that case, I can use `list(user_input)[0]`